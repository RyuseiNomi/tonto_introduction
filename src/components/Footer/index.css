.footer {
  height: 5em;
  display: flex;
  align-items: center;
  justify-content: center;
}

.footer p {
  font-size: 12pt;
}
