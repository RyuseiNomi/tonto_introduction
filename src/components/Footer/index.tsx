import React from 'react';
import './index.css';

function Footer() {

  return (
    <div className="footer">
      <p>©︎2022    <a href="https://twitter.com/27ma4_ton10" rel="noopener noreferrer" target="_blank">@27ma4_ton10</a></p>
    </div>
  );
};

export default Footer;
