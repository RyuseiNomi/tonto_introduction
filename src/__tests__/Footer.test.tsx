import { cleanup, act } from '@testing-library/react';
import { createRoot } from 'react-dom/client';
import renderer from 'react-test-renderer';
import Footer from './../components/Footer';

let container = null;
let root = null;
const sample_twitter_url = "http://sample_twitter_url.hoge";

beforeEach(() => {
  // container の定義
  container = document.createElement("div");
  document.body.appendChild(container);
  root = createRoot(container);
});

afterEach(cleanup);

// =============== snapshot test =================
it('Footer が表示されていること', () => {
  const component = renderer.create(<Footer twittr_url={sample_twitter_url} />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// ================ act tests ====================
it('Footer に西暦と twitter id が表示されていること', () => {
  act(() => {
    root.render(<Footer twitter_url={sample_twitter_url} />);
  })

  const displayed = document.querySelector("[data-testid=footer]");
  expect(displayed.textContent).toBe("Privacy Policy©︎2022    @27ma4_ton10_v1");
  console.log(displayed.textContent);
});
