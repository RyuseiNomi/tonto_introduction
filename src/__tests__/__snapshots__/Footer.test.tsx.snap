// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer が表示されていること 1`] = `
<div
  className="footer"
  data-testid="footer"
>
  <p
    className="privacy_policy"
    onClick={[Function]}
  >
    Privacy Policy
  </p>
  <p>
    ©︎2022    
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      @27ma4_ton10_v1
    </a>
  </p>
</div>
`;
